Userland/Libraries/LibWeb/Painting/StackingContext.cpp

```diff
@@ -310,12 +310,13 @@ void StackingContext::paint(PaintContext& context) const
         if (auto masking_area = paintable_box().get_masking_area(); masking_area.has_value()) {
             if (masking_area->is_empty())
                 return;
-            auto mask_bitmap = paintable_box().calculate_mask(context, *masking_area);
-            if (mask_bitmap) {
+            auto mask_and_clip_path_display_lists = paintable_box().calculate_mask(context, *masking_area);
+            auto mask_display_list = mask_and_clip_path_display_lists.mask_display_list;
+            if (!mask_display_list->is_empty()) {
```

Comment:
instead of checking whether display list is empty let's simply not allocate it when paintable does not have a mask.